File: DataStructure/src/DataStructure/Recusrsion/Factorial.java
package DataStructure.Recusrsion;

public class Factorial {

    // solution using recursion -- functional recursion
    public static int factorial(int n) {

        // base case
        if (n == 1) {
            return 1;
        }

        // releation
        return n * factorial(n - 1);

    }

    public static void main(String args[]) {

        int n = 6;
        System.out.println(factorial(n));
    }

}

File: JavascriptDSA/recursion/basic/factorial.js
// factorial of 2 using recursion

function factorialIterative(n) {
  let res = 1;

  for (let i = n; i > 1; i--) {
    res = res * i;
  }

  return res;
}

function recursiveSolution(n) {
  if (n === 1) {
    return 1;
  }

  return n * recursiveSolution(n - 1);
}

console.log(factorialIterative(5));
console.log(recursiveSolution(5));
